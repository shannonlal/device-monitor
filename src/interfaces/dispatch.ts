export interface IDispatch {
    dispatch?: (action: any) => Promise<any>;
}
