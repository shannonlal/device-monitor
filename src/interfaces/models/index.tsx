export interface IDeviceDetailsModel {
    name?: string;
    diskFree?: number;
    appVersion: string;
    appBuild: string;
    operatingSystem: string;
    osVersion: string;
    platform: 'ios' | 'android' | 'electron' | 'web';
    memUsed?: number;
    diskTotal?: number;
    model: string;
    manufacturer: string;
    uuid: string;
    isVirtual: boolean;
}

export interface IUser {
    eMail: string;
    firstName: string;
    lastName: string;
}

export interface IAuthorization {
    eMail: string;
    password: string;
}

export interface IWifiSummary {
    signalStrength: number;
    ssid: string;
}

// This wifi details interface will expand in the future.  Stubbed for now
export interface IWifiDetails {
    signalStrength: number;
    ssid: string;
    security: string;
}
