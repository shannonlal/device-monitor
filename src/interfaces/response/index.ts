export interface IResponse<T> {
    content?: T;
    httpStatus: number;
    errorMessage?: string;
}
