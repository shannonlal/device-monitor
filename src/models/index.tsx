export type DeviceDetailsModel = {
    name?: string;
    diskFree?: number;
    appVersion: string;
    appBuild: string;
    operatingSystem: string;
    osVersion: string;
    platform: 'ios' | 'android' | 'electron' | 'web';
    memUsed?: number;
    diskTotal?: number;
    model: string;
    manufacturer: string;
    uuid: string;
    isVirtual: boolean;
};
