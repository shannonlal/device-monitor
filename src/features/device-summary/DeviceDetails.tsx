import React from 'react';
import { IonText, IonRow, IonCol, IonGrid } from '@ionic/react';
import {DeviceDetailsModel} from '../../models';
import { DeviceField } from '../../components/common';

type DeviceDetailsProps = {
    headerLabel: string;
    headerField: string;
    deviceDetails: DeviceDetailsModel|undefined;
};
const DeviceDetails: React.FC<DeviceDetailsProps> = (props: DeviceDetailsProps): React.ReactElement => {
    const getDeviceDetail = (label: string, value: any, id: number) => {
        return <DeviceField label={label} value={value} key={`df-${id}-${label}-${value}`} />;
    };

    const getDeviceDetails = (details: DeviceDetailsModel| undefined) => {

        if( typeof details === 'undefined') {
            return;
        }

        const keys: string[] = Object.keys( details );
        if( details && keys.length > 0){

            const deviceDetails = (Object.keys(details) as Array<keyof typeof details>).map ( (field, i) => {
                return getDeviceDetail(field, details[field], i);
            })

            return deviceDetails;
        }
    };
    return (
        <IonGrid>
            <IonRow>
                <IonCol>
                    <IonText>{props.headerLabel}</IonText>
                </IonCol>
                <IonCol>
                    <IonText>{props.headerField}</IonText>
                </IonCol>
            </IonRow>
            {getDeviceDetails(props.deviceDetails)}
        </IonGrid>
    );
};

export default DeviceDetails;
