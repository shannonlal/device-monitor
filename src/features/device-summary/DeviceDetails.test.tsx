import React from 'react';
import { render } from '@testing-library/react';
import DeviceDetails from './DeviceDetails';
import {DeviceDetailsModel} from '../../interfaces/models';

describe('DeviceField', () => {
    test('rendering a simple device field', async () => {
        const headerLabel:string =  "Header";
        const headerField:string =  "Field";
        const deviceDetails:DeviceDetailsModel = {
            name: "MyPhone",
            diskFree: 101001,
            appVersion: '1.0.1',
            appBuild: '12.1.1',
            operatingSystem: 'Android',
            osVersion: 'OS',
            platform: 'ios',
            memUsed: 1322,
            diskTotal: 3113131,
            model: 'adadaa',
            manufacturer: 'Samsung',
            uuid: '12313231213123asdsdad',
            isVirtual: true
        }

        const { asFragment } = render(<DeviceDetails headerLabel={headerLabel} 
                                        headerField={headerField} 
                                        deviceDetails={deviceDetails} />);

        expect(asFragment()).toMatchSnapshot();
    }); 

    test('rendering a device details with optional field', async () => {
        const headerLabel:string =  "Header";
        const headerField:string =  "Field";
        const deviceDetails:DeviceDetailsModel = {
            name: "MyPhone",
            appVersion: '1.0.1',
            appBuild: '12.1.1',
            operatingSystem: 'Android',
            osVersion: 'OS',
            platform: 'ios',
            model: 'adadaa',
            manufacturer: 'Samsung',
            uuid: '12313231213123asdsdad',
            isVirtual: true
        }

        const { asFragment } = render(<DeviceDetails headerLabel={headerLabel} 
                                        headerField={headerField} 
                                        deviceDetails={deviceDetails} />);

        expect(asFragment()).toMatchSnapshot();
    }); 

    test('rendering a device details without device information', async () => {
        const headerLabel:string =  "Header";
        const headerField:string =  "Field";

        const { asFragment } = render(<DeviceDetails headerLabel={headerLabel} 
                                        headerField={headerField} />);

        expect(asFragment()).toMatchSnapshot();
    }); 

    test('rendering a device details without device information', async () => {
        const headerLabel:string =  "Header";
        const headerField:string =  "Field";

        const { asFragment } = render(<DeviceDetails headerLabel={headerLabel} 
                                        headerField={headerField} deviceDetails ={undefined}/>);

        expect(asFragment()).toMatchSnapshot();
    }); 
})