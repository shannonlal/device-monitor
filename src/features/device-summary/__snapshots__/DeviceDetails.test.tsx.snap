// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceField rendering a device details with optional field 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Header
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Field
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          MyPhone
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          appVersion
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          1.0.1
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          appBuild
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          12.1.1
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          operatingSystem
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Android
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          osVersion
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          OS
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          platform
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          ios
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          model
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          adadaa
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          manufacturer
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Samsung
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          uuid
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          12313231213123asdsdad
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          isVirtual
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text />
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`DeviceField rendering a device details without device information 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Header
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Field
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`DeviceField rendering a device details without device information 2`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Header
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Field
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`DeviceField rendering a simple device field 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Header
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Field
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          MyPhone
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          diskFree
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          101001
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          appVersion
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          1.0.1
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          appBuild
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          12.1.1
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          operatingSystem
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Android
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          osVersion
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          OS
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          platform
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          ios
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          memUsed
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          1322
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          diskTotal
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          3113131
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          model
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          adadaa
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          manufacturer
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Samsung
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          uuid
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          12313231213123asdsdad
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          isVirtual
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text />
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;
